afpacket: panic running the example code on i386

I am trying to run https://github.com/google/gopacket/blob/master/examples/afpacket/afpacket.go As soon as the first packet arrives the program crashes with a panic at the `ZeroCopyReadPacketData()` call:
```
root@appl1:/home/maintenance/workspace# ./afpacket
2018/09/11 00:41:39 Starting on interface "eth0"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8049e8c]

goroutine 1 [running]:
runtime/internal/atomic.Xadd64(0xa08613c, 0x1, 0x0, 0xffffffff, 0x1)
        /usr/local/go/src/runtime/internal/atomic/asm_386.s:102 +0xc
github.com/google/gopacket/afpacket.(*TPacket).pollForFirstPacket(0xa0860b0, 0x81de0c0, 0xa08611c, 0x0, 0x0)
        /root/go/src/github.com/google/gopacket/afpacket/afpacket.go:468 +0xfd
github.com/google/gopacket/afpacket.(*TPacket).ZeroCopyReadPacketData(0xa0860b0, 0x81bf378, 0xa0860b0, 0xa0860b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /root/go/src/github.com/google/gopacket/afpacket/afpacket.go:308 +0x9f
main.main()
        /home/maintenance/workspace/afpacket.go:55 +0x25f
```

My environment:
```
go version go1.11 linux/386
```
```
Linux appl1 4.9.0-8-686 #1 SMP Debian 4.9.110-3+deb9u4 (2018-08-21) i686 GNU/Linux
```
```
GOARCH="386"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="386"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_386"
GCCGO="gccgo"
GO386="sse2"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m32 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build584579405=/tmp/go-build -gno-record-gcc-switches"
```
I do not know if it is useful but here is the content of `/proc/cpuinfo`
```
processor       : 0
vendor_id       : CentaurHauls
cpu family      : 6
model           : 13
model name      : VIA Eden Processor  500MHz
stepping        : 0
cpu MHz         : 498.687
cache size      : 128 KB
physical id     : 0
siblings        : 1
core id         : 0
cpu cores       : 1
apicid          : 0
initial apicid  : 0
fdiv_bug        : no
f00f_bug        : no
coma_bug        : no
fpu             : yes
fpu_exception   : yes
cpuid level     : 1
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge cmov pat clflush acpi mmx fxsr sse sse2 tm nx pni est tm2 xtpr rng rng_en ace ace_en ace2 ace2_en phe phe_en pmm pmm_en
bugs            : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf
bogomips        : 997.37
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 32 bits virtual
power management:
```

I am not completely sure if the panic is related to gopacket at all but i am not very experienced in go.

--- a/afpacket/afpacket.go
+++ b/afpacket/afpacket.go
@@ -103,6 +103,8 @@ func (s *SocketStatsV3) QueueFreezes() uint {
 
 // TPacket implements packet receiving for Linux AF_PACKET versions 1, 2, and 3.
 type TPacket struct {
+	// stats is simple statistics on TPacket's run. This MUST be the first entry to ensure alignment for sync.atomic
+	stats Stats
 	// fd is the C file descriptor.
 	fd int
 	// ring points to the memory space of the ring buffer shared by tpacket and the kernel.
@@ -130,8 +132,6 @@ type TPacket struct {
 	v3 v3wrapper
 
 	statsMu sync.Mutex // guards stats below
-	// stats is simple statistics on TPacket's run.
-	stats Stats
 	// socketStats contains stats from the socket
 	socketStats SocketStats
 	// same as socketStats, but with an extra field freeze_q_cnt

--- a/pcap/pcap.go
+++ b/pcap/pcap.go
@@ -154,6 +154,10 @@ const bpfInstructionBufferSize = 8 * MaxBpfInstructions
 //
 // Handles are already pcap_activate'd
 type Handle struct {
+	// stop is set to a non-zero value by Handle.Close to signal to
+	// getNextBufPtrLocked to stop trying to read packets
+	// This must be the first entry to ensure alignment for sync.atomic
+	stop uint64
 	// cptr is the handle for the actual pcap C object.
 	cptr        *C.pcap_t
 	timeout     time.Duration
@@ -161,9 +165,6 @@ type Handle struct {
 	deviceIndex int
 	mu          sync.Mutex
 	closeMu     sync.Mutex
-	// stop is set to a non-zero value by Handle.Close to signal to
-	// getNextBufPtrLocked to stop trying to read packets
-	stop uint64
 
 	// Since pointers to these objects are passed into a C function, if
 	// they're declared locally then the Go compiler thinks they may have

